### PR TITLE
json_query, no more 'unkown type' errors

### PR DESCRIPTION
**Pull request by bcoca**
Added missing types for list and dicts, previously we had fixed the 'strings' issue

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

filters/json_query

### CODE REVIEW

**Comment by bcoca** [comment]
cc @mkrizek

File: changelogs/fragments/json_query_more_types.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Avoid 'unkown type' errors for more Ansible internal types.
```
Comment (felixfontein) [inline changelogs/fragments/json_query_more_types.yml:2]:
```suggestion
  - json_query filter plugin - avoid 'unkown type' errors for more Ansible internal types (https://github.com/ansible-collections/community.general/pull/2607).
```

File: plugins/filter/json_query.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ def json_query(data, expr):
     # Hack to handle Ansible String Types
     # See issue: https://github.com/ansible-collections/community.general/issues/320
     jmespath.functions.REVERSE_TYPES_MAP['string'] = jmespath.functions.REVERSE_TYPES_MAP['string'] + ('AnsibleUnicode', 'AnsibleUnsafeText', )
+    jmespath.functions.REVERSE_TYPES_MAP['string'] = jmespath.functions.REVERSE_TYPES_MAP['list'] + ('AnsibleSequence', )
+    jmespath.functions.REVERSE_TYPES_MAP['string'] = jmespath.functions.REVERSE_TYPES_MAP['dict'] + ('AnsibleMapping', )
```
Comment (felixfontein) [inline plugins/filter/json_query.py:42]:
```suggestion
    jmespath.functions.REVERSE_TYPES_MAP['list'] = jmespath.functions.REVERSE_TYPES_MAP['list'] + ('AnsibleSequence', )
    jmespath.functions.REVERSE_TYPES_MAP['dict'] = jmespath.functions.REVERSE_TYPES_MAP['dict'] + ('AnsibleMapping', )
```

File: changelogs/fragments/json_query_more_types.yml (hunk 1 above)
Comment (Akasurde) [inline changelogs/fragments/json_query_more_types.yml:2]:
```suggestion
  - json_query filter plugin - avoid 'unknown type' errors for more Ansible internal types (https://github.com/ansible-collections/community.general/pull/2607).
```

**Review by Akasurde — APPROVED** [review]
LGTM

File: plugins/filter/json_query.py [hunk 3]
```diff
@@ -35,9 +35,11 @@ def json_query(data, expr):
         raise AnsibleError('You need to install "jmespath" prior to running '
                            'json_query filter')
 
-    # Hack to handle Ansible String Types
+    # Hack to handle Ansible Unsafe text, AnsibleMapping and AnsibleSequence
     # See issue: https://github.com/ansible-collections/community.general/issues/320
     jmespath.functions.REVERSE_TYPES_MAP['string'] = jmespath.functions.REVERSE_TYPES_MAP['string'] + ('AnsibleUnicode', 'AnsibleUnsafeText', )
+    jmespath.functions.REVERSE_TYPES_MAP['list'] = jmespath.functions.REVERSE_TYPES_MAP['array'] + ('AnsibleSequence', )
+    jmespath.functions.REVERSE_TYPES_MAP['dict'] = jmespath.functions.REVERSE_TYPES_MAP['object'] + ('AnsibleMapping', )
```
[the remark this answers is not in the document]
Comment (Akasurde) [reply plugins/filter/json_query.py]: Yes. my bad.

**Comment by felixfontein** [comment]
I think this code should eventually be reactored to only add these strings when they aren't already in the tuple, since right now during every invocation, the tuples will grow.

Also having some tests for these specific behaviors would be good to avoid regressions.

In any case, I'll merge this for now.

**Comment by felixfontein** [comment]
@bcoca @Akasurde thanks for fixing this!

**Pull request merged**